br compact backup user document (#2676)
pingcap/docs-tidb-operator: zh/backup-restore-cr.md

* `.spec.local.volume`：持久卷配置。
* `.spec.local.volumeMount`：持久卷挂载配置。

## CompactBackup CR 字段介绍

对于 TiDB v9.0.0 及以上版本的集群，你可以使用 `CompactBackup` 加速日志恢复。要将日志备份数据压缩为结构化 SST 文件，你可以通过创建一个自定义的 `CompactBackup` CR 对象来描述一次备份任务。以下是 `CompactBackup` CR 各个字段的具体含义：

* `.spec.startTs`：指定日志压缩备份的起始时间戳。
* `.spec.endTs`：指定日志压缩备份的结束时间戳。
* `.spec.concurrency`：指定同时进行的压缩日志任务的最大数量，默认值为 `4`。
* `.spec.maxRetryTimes`：指定压缩任务失败的最大重试次数，默认值为 `6`。
* `.spec.toolImage`：指定 `CompactBackup` 使用的工具镜像。在 `CompactBackup` 中，唯一使用的工具镜像为 BR。使用 BR 备份时，你可以使用该字段指定 BR 的版本：

    - 如果未指定或者为空，默认使用镜像 `pingcap/br:${tikv_version}` 进行备份。

>     - 如果指定了 BR 的版本，例如 `.spec.toolImage: pingcap/br:v5.3.0`，那么使用指定的版本镜像进行备份。

Reviewer: ```suggestion
    - 如果指定了 BR 的版本，例如 `.spec.toolImage: pingcap/br:v8.5.0`，那么使用指定的版本镜像进行备份。
```

Reviewer 2: @RidRisR PTAL

    - 如果指定了镜像但未指定版本，例如 `.spec.toolImage: private/registry/br`，那么使用镜像 `private/registry/br:${tikv_version}` 进行备份。
* `.spec.env`：指定运行压缩备份任务的 Pod 的环境变量信息。

> * `.spec.affinity`：指定运行备份任务的 Pod 亲和性 (affinity) 配置。关于亲和性的详细说明，请参阅[亲和性与反亲和性 ](https://kubernetes.io/zh-cn/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity)。

Author: ```suggestion
* `.spec.affinity`：指定运行备份任务的 Pod 亲和性 (affinity) 配置。关于亲和性的详细说明，请参阅[亲和性与反亲和性](https://kubernetes.io/zh-cn/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity)。
```

* `.spec.local.volume`：持久卷配置。
* `.spec.local.volumeMount`：持久卷挂载配置。

## CompactBackup CR 字段介绍

> 对于 ***v9.0.0*** 以上的TiDB集群，你可以使用 `CompactBackup` 加速日志恢复。为了将日志备份数据压缩成结构化SST，用户可以通过创建一个自定义的 `CompactBackup` CR 对象来描述一次备份，以下介绍 `CompactBackup` CR 各个字段的具体含义。

Reviewer 2: ```suggestion
对于 TiDB v9.0.0 及以上版本的集群，你可以使用 `CompactBackup` 加速日志恢复。要将日志备份数据压缩为结构化 SST 文件，你可以通过创建一个自定义的 `CompactBackup` CR 对象来描述一次备份任务。以下是 `CompactBackup` CR 各个字段的具体含义：
```

Author: 那英文部分是不是也要删掉强调？

> * `.spec.startTs`: 用于指定日志压缩备份的的范围起始时间戳。
> * `.spec.endTs`: 用于指定日志压缩备份的范围结束时间戳。
> * `.spec.concurrency`: 最大同时进行的压缩日志任务的数量。默认为4
> * `.spec.maxRetryTimes`: 压缩任务失败的最大重试次数。默认为6

Reviewer 2: ```suggestion
* `.spec.startTs`：指定日志压缩备份的起始时间戳。
* `.spec.endTs`：指定日志压缩备份的结束时间戳。
* `.spec.concurrency`：指定同时进行的压缩日志任务的最大数量，默认值为 `4`。
* `.spec.maxRetryTimes`：指定压缩任务失败的最大重试次数，默认值为 `6`。
```

> * `.spec.toolImage`：用于指定 `CompactBackup` 使用的工具镜像。在 `CompactBackup` 中，唯一使用的工具镜像为 BR

Reviewer 2: ```suggestion
* `.spec.toolImage`：指定 `CompactBackup` 使用的工具镜像。在 `CompactBackup` 中，唯一使用的工具镜像为 BR。使用 BR 备份时，你可以使用该字段指定 BR 的版本：
```

>     - 使用 BR 备份时，可以用该字段指定 BR 的版本:
>         - 如果未指定或者为空，默认使用镜像 `pingcap/br:${tikv_version}` 进行备份。
>         - 如果指定了 BR 的版本，例如 `.spec.toolImage: pingcap/br:v5.3.0`，那么使用指定的版本镜像进行备份。
>         - 如果指定了镜像但未指定版本，例如 `.spec.toolImage: private/registry/br`，那么使用镜像 `private/registry/br:${tikv_version}` 进行备份。

Reviewer 2: ```suggestion
    - 如果未指定或者为空，默认使用镜像 `pingcap/br:${tikv_version}` 进行备份。
    - 如果指定了 BR 的版本，例如 `.spec.toolImage: pingcap/br:v5.3.0`，那么使用指定的版本镜像进行备份。
    - 如果指定了镜像但未指定版本，例如 `.spec.toolImage: private/registry/br`，那么使用镜像 `private/registry/br:${tikv_version}` 进行备份。
```

* `.spec.env`：指定运行压缩备份任务的 Pod 的环境变量信息。

> * `.spec.affinity`：指定运行备份任务的 Pod 亲和性配置，关于 affinity 的使用说明，请参阅 [Affinity & AntiAffinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity)。
> * `.spec.tolerations`：指定运行压缩备份任务的 Pod 能够调度到带有与之匹配的[污点](https://kubernetes.io/docs/reference/glossary/?all=true#term-taint) (Taint) 的节点上。关于污点与容忍度的更多说明，请参阅 [Taints and Tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)。
> * `.spec.podSecurityContext`：指定运行压缩备份任务的 Pod 的安全上下文配置，允许 Pod 以非 root 用户的方式运行，关于 podSecurityContext 的更多说明，请参阅[以非 root 用户运行容器](containers-run-as-non-root-user.md)。
> * `.spec.priorityClassName`：指定运行压缩备份任务的 Pod 的 priorityClass 的名称，以设置运行优先级，关于 priorityClass 的更多说明，请参阅 [Pod Priority and Preemption](https://kubernetes.io/zh/docs/concepts/scheduling-eviction/pod-priority-preemption/)。
> * `.spec.imagePullSecrets`：指定运行压缩备份任务的 Pod 的 [imagePullSecrets](https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod)
> * `.spec.serviceAccount`：指定恢复时所使用的 ServiceAccount 名称。

Reviewer 2: ```suggestion
* `.spec.affinity`：指定运行备份任务的 Pod 亲和性 (affinity) 配置。关于亲和性的详细说明，请参阅[亲和性与反亲和性 ](https://kubernetes.io/zh-cn/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity)。
* `.spec.tolerations`：指定运行压缩备份任务的 Pod 能够被调度到带有与之匹配的[污点 (Taint)](https://kubernetes.io/docs/reference/glossary/?all=true#term-taint) 的节点上。关于污点与容忍度的更多说明，请参阅[污点和容忍度](https://kubernetes.io/zh-cn/docs/concepts/scheduling-eviction/taint-and-toleration/)。
* `.spec.podSecurityContext`：指定运行压缩备份任务的 Pod 的安全上下文配置，以支持非 root 用户运行 Pod。关于 `podSecurityContext` 的更多说明，请参阅[以非 root 用户运行容器](containers-run-as-non-root-user.md)。
* `.spec.priorityClassName`：指定运行压缩备份任务的 Pod 的 `priorityClass` 的名称，用于设置运行优先级。关于 `priorityClass` 的更多说明，请参阅 [Pod 优先级和抢占](https://kubernetes.io/zh-cn/docs/concepts/scheduling-eviction/pod-priority-preemption/)。
* `.spec.imagePullSecrets`：指定运行压缩备份任务的 Pod 使用的 [`imagePullSecrets`](https://kubernetes.io/zh-cn/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod)。
* `.spec.serviceAccount`：指定恢复时使用的 ServiceAccount 名称。
```

* `.spec.useKMS`：指定恢复时是否使用 AWS-KMS 解密备份使用的 S3 存储密钥。

> * `.spec.br`：BR 相关配置，具体介绍参考 [BR 字段介绍](#br-字段介绍)。
> * `.spec.s3`：S3 兼容存储相关配置，具体介绍参考 [S3 字段介绍](#s3-存储字段介绍)。
> * `.spec.gcs`：GCS 存储相关配置，具体介绍参考 [GCS 字段介绍](#gcs-存储字段介绍)。
> * `.spec.azblob`：Azure Blob Storage 存储相关配置，具体介绍参考 [Azure Blob Storage 字段介绍](#azure-blob-storage-存储字段介绍)。

Reviewer 2: ```suggestion
* `.spec.br`：BR 相关配置，详情请参阅 [BR 字段介绍](#br-字段介绍)。
* `.spec.s3`：S3 兼容存储相关配置，详情请参阅 [S3 字段介绍](#s3-存储字段介绍)。
* `.spec.gcs`：GCS 存储相关配置，详情请参阅 [GCS 字段介绍](#gcs-存储字段介绍)。
* `.spec.azblob`：Azure Blob Storage 存储相关配置，详情请参阅 [Azure Blob Storage 字段介绍](#azure-blob-storage-存储字段介绍)。
```

+ `backupTemplate`：快照备份相关配置。指定快照备份集群及远程存储相关的配置，字段和 Backup CR 中的 `spec` 一样，详细介绍可参考 [Backup CR 字段介绍](#backup-cr-字段介绍)。
+ `logBackupTemplate`：日志备份相关配置。指定日志备份集群及远程存储相关的配置，字段和 Backup CR 中的 `spec` 一样，详细介绍可参考 [Backup CR 字段介绍](#backup-cr-字段介绍)，日志备份随 `backupSchedule` 创建、删除，且根据 `.spec.maxReservedTime` 进行回收。日志备份名称在 `status.logBackup` 中保存。

> + `compactBackupTemplate`: 压缩日志备份的配置模板，字段和 CompactBackup CR 中的 `spec` 一样，详细介绍可参考 [ComapctBackup CR 字段介绍](#compactbackup-cr-字段介绍 )，压缩日志备份随 `backupSchedule` 创建、删除，日志备份名称在 `status.logBackup` 中保存。压缩日志备份的存储设置应与同一 `backupSchedule` 中的 `logBackupTemplate` 相同

Reviewer 2: ```suggestion
+ `compactBackupTemplate`：压缩日志备份的配置模板，字段和  CompactBackup CR 中的 `spec` 一样，详细介绍可参考 [CompactBackup CR 字段介绍](#compactbackup-cr-字段介绍)。压缩日志备份会随 `backupSchedule` 创建和删除，日志备份名称存储在 `status.logBackup` 中。压缩日志备份的存储设置应与同一 `backupSchedule` 中的 `logBackupTemplate` 保持一致。
```

    + `.spec.maxBackups`：一种备份保留策略，决定定时备份最多可保留的备份个数。超过该数目，就会将过时的备份删除。如果将该项设置为 `0`，则表示保留所有备份。
    + `.spec.maxReservedTime`：一种备份保留策略，按时间保留备份。例如将该参数设置为 `24h`，表示只保留最近 24 小时内的备份条目。超过这个时间的备份都会被清除。时间设置格式参考 [`func ParseDuration`](https://golang.org/pkg/time/#ParseDuration)。如果同时设置 `.spec.maxBackups` 和 `.spec.maxReservedTime`，则以 `.spec.maxReservedTime` 为准。
    + `.spec.schedule`：Cron 的时间调度格式。具体格式可参考 [Cron](https://en.wikipedia.org/wiki/Cron)。

>     + `.spec.compactInterval`: 用于触发新压缩任务的时间间隔。

Reviewer 2: ```suggestion
    + `.spec.compactInterval`：用于触发新压缩任务的时间间隔。
```
